sound/soc/intel/boards/sof_sdw.c

```diff
@@ -399,22 +399,27 @@ static struct sof_sdw_codec_info codec_info_list[] = {
 	},
 };
 
-static inline int find_codec_info_part(u64 adr)
+static inline int find_codec_info_part(u64 adr, const char *dai_name)
 {
 	unsigned int part_id, sdw_version;
 	int i;
 
 	part_id = SDW_PART_ID(adr);
 	sdw_version = SDW_VERSION(adr);
-	for (i = 0; i < ARRAY_SIZE(codec_info_list); i++)
+	for (i = 0; i < ARRAY_SIZE(codec_info_list); i++) {
 		/*
 		 * A codec info is for all sdw version with the part id if
 		 * version_id is not specified in the codec info.
 		 */
 		if (part_id == codec_info_list[i].part_id &&
 		    (!codec_info_list[i].version_id ||
-		     sdw_version == codec_info_list[i].version_id))
-			return i;
+		     sdw_version == codec_info_list[i].version_id)) {
+			if (!dai_name || !strcmp(dai_name, "") ||
+			    !strncmp(dai_name, codec_info_list[i].dai_name, 64)) {
```

Comment:
why strncmp? we only want to match the first 64 bytes?

Reply:
> why strncmp? we only want to match the first 64 bytes?

Because `dai_name` is `const char dai_name[64]`

Reply:
this could be
```
	if (part_id == codec_info_list[i].part_id &&
	    (!codec_info_list[i].version_id ||
	     sdw_version == codec_info_list[i].version_id) &&
	    (!dai_name || !strcmp(dai_name, "") ||
	    !strncmp(dai_name, codec_info_list[i].dai_name, 64)))
		return i;
```

Reply:
> this could be
> 
> ```
> 	if (part_id == codec_info_list[i].part_id &&
> 	    (!codec_info_list[i].version_id ||
> 	     sdw_version == codec_info_list[i].version_id) &&
> 	    (!dai_name || !strcmp(dai_name, "") ||
> 	    !strncmp(dai_name, codec_info_list[i].dai_name, 64)))
> 		return i;
> ```

Updated. Thanks.